lib/zclient.c

```diff
@@ -1415,20 +1495,47 @@ int zapi_nexthop_from_nexthop(struct zapi_nexthop *znh,
 
 	znh->type = nh->type;
 	znh->vrf_id = nh->vrf_id;
+	znh->weight = nh->weight;
 	znh->ifindex = nh->ifindex;
 	znh->gate = nh->gate;
 
 	if (nh->nh_label && (nh->nh_label->num_labels > 0)) {
+
+		/* Validate */
+		if (nh->nh_label->num_labels > MPLS_MAX_LABELS)
```

Comment:
some error message not needed for troubleshooting purposes?

Reply:
I believe there's an error log in the caller, if this fails. It made more sense to do it there, where there's more info / context?